geometric_tools.py

```diff
@@ -0,0 +1,221 @@
+        """Find the closest point of the boundaries of our truncated cone to the point passed as second parameter"""
+        # We get back the list of vertices
+        point_list = list(self.arc.coords)
+        
+        # We compute the projection on the first half line
+        minimum = [find_projection_half_line(point_to_tuple(self.limit_points[0]), (0, 0), point)]
+        minimum.append(distance_tuple(point, minimum[0]))
+        
+        # And the second
+        proj = find_projection_half_line(point_to_tuple(self.limit_points[1]), (0, 0), point)
+        dist = distance_tuple(point, proj)
+        if dist < minimum[1]:
+                minimum = [proj, dist]
+        
+        # For each segment within the list we check the projection
+        for i in range(len(point_list) - 1):
+            first = point_list[i]
+            second = point_list[i + 1]
+            
+            proj = find_projection_segment(first, second, point)
+            dist = distance_tuple(point, proj)
+            
+            if dist < minimum[1]:
+                minimum = [proj, dist]
+        
+        # We return the result
+        return S.Point(minimum[0][0], minimum[0][1])
+        
+
+def in_half_plane(origin, orthogonal, point):
+    if (point.x - origin.x) * orthogonal.x + (point.y - origin.y) * orthogonal.y >= -0.000001:
+        return True
+    return False
+   
+    
+def intersection_line_line(origin1, ortho, point1, point2, vmax):
+    """This function makes an intersection between two lines with the following representations:
+    line1 : origin1 + ortho (vector that is normal to the line)
+    line2 : two points
+    Return the list of intersection points (empty or of cardinal 1)
+    """
+    # Direction vector of the second line
+    vect1 = S.Point(point1.x - point2.x, point1.y - point2.y)
+    # Test if the lines are colinear or not
+    if vect1.x * ortho.x + vect1.y * ortho.y == 0:
+        return []
+    # Find a second point on the first line
+    origin2 = S.Point(origin1.x + ortho.y, origin1.y - ortho.x)
+    # Compute the intersection with an ugly forumla
+    px = (origin1.x * origin2.y - origin1.y * origin2.x) * (point1.x - point2.x) - (point1.x * point2.y - point1.y * point2.x) * (origin1.x - origin2.x)
+    px /= (origin1.x - origin2.x) * (point1.y - point2.y) - (origin1.y - origin2.y) * (point1.x - point2.x)
+    py = (origin1.y * origin2.x - origin1.x * origin2.y) * (point1.y - point2.y) - (point1.y * point2.x - point1.x * point2.y) * (origin1.y - origin2.y)
+    py /= (origin1.y - origin2.y) * (point1.x - point2.x) - (origin1.x - origin2.x) * (point1.y - point2.y)
+    # Test if our intersection is within our range or not
+    if abs(px) <= vmax and abs(py) <= vmax:
+        return [S.Point(px, py)]
+    else:
+        return []
+
+    
+# TODO : Find a way to not bug if the two individual are in colision at the begining. (I.e. : if there is a bug at the begining then find a way to correct it in the end)
```

Comment:
I leave prefer to leave it here for now. This is for a special case that is not suposed to happen. In the end I wil have to change some things (ex: take environment into account) so I could correct it then. Moreover I don't know yet how to deal with the problem (will have to talk with Johanna).